api/coreservice_with_height.go

```diff
@@ -0,0 +1,71 @@
+package api
+
+import (
+	"context"
+
+	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/iotexproject/iotex-address/address"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/iotexproject/iotex-core/v2/action"
+	"github.com/iotexproject/iotex-core/v2/blockchain/genesis"
+	"github.com/iotexproject/iotex-core/v2/pkg/log"
+	"github.com/iotexproject/iotex-core/v2/pkg/tracer"
+	"github.com/iotexproject/iotex-core/v2/pkg/util/byteutil"
+)
+
+type (
+	// CoreServiceReaderWithHeight is an interface for state reader at certain height
+	CoreServiceReaderWithHeight interface {
+		Account(address.Address) (*iotextypes.AccountMeta, *iotextypes.BlockIdentifier, error)
+		ReadContract(context.Context, address.Address, action.Envelope) (string, *iotextypes.Receipt, error)
+	}
+
+	coreServiceReaderWithHeight struct {
+		cs     *coreService
+		height uint64
+	}
+)
+
+func newCoreServiceWithHeight(cs *coreService, height uint64) *coreServiceReaderWithHeight {
+	return &coreServiceReaderWithHeight{
+		cs:     cs,
+		height: height,
+	}
+}
+
+func (core *coreServiceReaderWithHeight) Account(addr address.Address) (*iotextypes.AccountMeta, *iotextypes.BlockIdentifier, error) {
+	if !core.cs.archiveSupported {
```

Comment:
https://github.com/iotexproject/iotex-core/pull/4529/commits/bd7ca1f4172ba460eee146c3de91a5e5f3bc78ea#diff-699e59a265a0599d1a49698bbf378f57a7aeb60957441a257249102ac8785a8dR50 will return error if archive is not supported

Reply:
it is already supported by `factory/statedb`, the actual archive support will be enabled in #4517, #4518